src/v/kafka/server/tests/alter_user_scram_credentials_test.cc

```diff
@@ -609,3 +609,242 @@ FIXTURE_TEST(
     BOOST_CHECK_EQUAL(
       resp.data.results[0].error_code, kafka::error_code::duplicate_resource);
 }
+
+FIXTURE_TEST(
```

Comment:
Should there be a test for when authentication is not enabled on the cluster?

I.e., I think you're missing a condition check; if authentication is not enabled, alter/delete a SCRAM credential for a superuser is not possible.